### ISSUE TITLE
Go over non-implemented LTER checks and look for any good ones

### ISSUE THREAD

**Issue by amoeba** [NCEAS/mdqengine#41]
@gothub may have done this but @mbjones was curious if any of the non-implemented LTER checks could be useful. The last time I looked at that sheet, I wasn't really looking at the other checks that closely. I'll take a look and report back. @gothub did you find any?

**Comment by gothub** [comment]
@amoeba Yeah, good idea to check. Seems like the priority 1 items that haven't been implemented we should at least consider:

- [ ] pastDoiCorrect
   - Check that a pre-generated DOI matches what the expected DOI will be
- [ ] integrityChecksum
   - check that a checksum in the L0 for entity matches the one pasta generates
- [ ] whitespaceLeadingTrailingPresent
   - check selected elements for leading or trailing whitespace
- [ ] dateFormatNonPreferred
   - Date format in metadata is a non-preferred format [LIST], and data matches
- [ ] dateFormatPreferred
   - Date format in metadata is a preferred format [LIST], and data matches
- [ ] numericFields
   - Numeric fields do not have string content

A couple of these are PASTA centric, so maybe not relevant. The others are "is the data
entered correctly", so maybe not so important and wouldn't score as high as other checks.

**Comment by amoeba** [comment]
I agree that these are pretty PASTA-specific and aren't very re-usable as-is. Some I picked out:

- `jagedArray`: I was thinking this is a really useful check... if your tabular dataset is ragged, send out a warning
- `dataWithinEnumeratedDomain`: This seems very useful.
- `numericFields`: Numeric field do not have string content
- `headerRowAttributeNames`: This seems very useful.  Is this the same as the already-implemented `tooManyFields`?
- Check that DOIs resolve

Other ideas coming out of reading this list,

- Check if filenames match format ids. This would be just a warning or info check. If an entity was supposed to be a ZIP you'd like to see a .zip file extension though it isn't a dealbreaker


Getting some good ideas for helper methods though...

- A function to check if a character vector has whitespace padding either side of it ("    asdf    ") on any element
- A function to check date formats
- A function to check that there are no duplicates in a set



I gotta run but didn't quite finish taking a look through.
